src/prototype_creep_harvest.js

```diff
@@ -54,36 +74,33 @@ Creep.prototype.spawnCarry = function() {
     }
   };
 
-  // Spawn carry
-  var energies = this.pos.lookFor(LOOK_ENERGY);
-  if (energies.length === 0) {
-    const containers = this.pos.findInRangePropertyFilter(FIND_STRUCTURES, 0, 'structureType', [STRUCTURE_CONTAINER], false, {
-      // TODO hardcoded for now, half of the container? Good idea?
-      filter: container => container.store.energy >= 1000
-    });
-    if (containers.length === 0) {
-      return false;
-    }
+  let energyAtPosition = 0;
+  var energies = this.pos.lookFor(LOOK_RESOURCES);
+  for (let energy of energies) {
+    energyAtPosition += energy.amount;
   }
 
-  if (energies.length > 0 && energies[0].amount < 50) {
-    return false;
+  if (energyAtPosition > carryCapacity) {
+    Game.rooms[this.memory.base].checkRoleToSpawn('carry', config.carry.maxPerTargetPerRoom, this.memory.routing.targetId, this.memory.routing.targetRoom);
+    this.memory.wait = waitTime;
+    return true;
   }
 
-  if (this.inBase()) {
-    if (energies.length > 0 && energies[0].amount < energyThreshold) {
-      return false;
+  let containers = this.pos.findInRange(FIND_STRUCTURES, 0, {
+    filter: function(object) {
+      if (object.structureType != STRUCTURE_CONTAINER) {
+        return false;
+      }
+      return true;
     }
+  });
+
+  for (let container of containers) {
+    energyAtPosition += _.sum(container.store);
   }
 
-  if (!existInArray(Game.rooms[this.memory.base].memory.queue, spawn)) {
-    if (typeof(this.memory.wait) === 'undefined') {
-      this.memory.wait = 0;
-    }
-    if (this.memory.wait <= 0) {
-      Game.rooms[this.memory.base].checkRoleToSpawn('carry', config.carry.maxPerTargetPerRoom, this.memory.routing.targetId, this.memory.routing.targetRoom);
-      this.memory.wait = waitTime;
-    }
+  if (energyAtPosition > carryCapacity) {
+    Game.rooms[this.memory.base].checkRoleToSpawn('carry', config.carry.maxPerTargetPerRoom, this.memory.routing.targetId, this.memory.routing.targetRoom);
```

Comment:
Even as I have CPU limit problems now, I still prever neat code more)) Does it really makes difference once per waitTime?